ICSharpCode.Decompiler/CSharp/Transforms/IntroduceUnsafeModifier.cs

```diff
@@ -142,7 +142,7 @@ public override bool VisitIdentifierExpression(IdentifierExpression identifierEx
 		public override bool VisitStackAllocExpression(StackAllocExpression stackAllocExpression)
 		{
 			base.VisitStackAllocExpression(stackAllocExpression);
-			return true;
+			return stackAllocExpression.GetResolveResult().Type is PointerType;
```

Comment:
Don't forget the return value of the base call.
It's theoretically possible to have a safe stackalloc that uses unsafe code in the size computation.